_el/exercises.md

```diff
@@ -336,172 +204,52 @@ piece of code with the output it produces".
+
+Οι [[Keun2016a](#CITE),[Keun2016b](#CITE)] ταξινόμησαν τα μηνύματα που παράγουν 69 εργαλεία αυτόματης βαθμολόγησης. Διαπίστωσαν ότι αυτές συχνά δεν δίνουν ανατροφοδότηση στο πώς να διορθωθούν τα προβλήματα ώστε να γίνει το επόμενο βήμα. Βρήκαν επίσης ότι οι περισσότεροι εκπαιδευτικοί δεν μπορούν εύκολα να προσαρμόσουν περισσότερα από τα εργαλεία στις ανάγκες τους - όπως και πολλοί άλλα εργαλεία ροής εργασίας, τείνουν να επιβάλλουν τις μη ευσταθείς υποθέσεις των δημιουργών τους σχετικά με τον τρόπο λειτουργίας των εκπαιδευτικών οργανισμών. Οι συγγραφείς των άρθρων αυτών συνεχίζουν το έργο τους σε αυτό το χώρο και το λεπτομερές σύστημα ταξινόμησης τους είναι πολύ χρήσιμο για τον εντοπισμό των κατάλληλων εργαλείων αυτόματης βαθμολόγησης.
+
+Οι [[Srid2016](#CITE)] συζητούν στρατηγικές ανταλλαγής σχολίων με τους μαθητές κατά τη διάρκεια της αυτόματης βαθμολόγησής τους. Μια πρώτη στρατηγική είναι να δίνεται το αναμενόμενο αποτέλεσμα των δοκιμών -- αλλά στη συνέχεια οι μαθητές μπορούν να ενσωματώσουν αυτό το αποτέλεσμα στον κώδικά τους (γιατί ότι μπορεί να λειτουργήσει ως εύκολη λύση, θα γίνει). Μια εναλλακτική είναι η αναφορά των αποτελεσμάτων επιτυχίας / αποτυχίας για τον κώδικα των εκπαιδευομένων, αλλά να δίνονται τα πραγματικά αποτελέσματα των δοκιμνών (ως είσοδος / έξοδος του προγράμματος) μετά την καταλληκτική ημερομηνία υποβολής. Αυτό μπορεί να είναι απογοητευτικό, διότι τους λέει στους μαθητές ότι είναι λάθος, αλλά όχι το γιατί είναι.
+
+Μια τρίτη επιλογή είναι να χρησιμοποιήσετε μια τεχνική που ονομάζεται [κατακερματισμός](#g:hashing) για τη δημιουργία μιας τιμής που εξαρτάται από το πραγματικό αποτέλεσμα αλλά δεν το αποκαλύπτει. Αν ο χρήστης παράγει ακριβώς την ίδια έξοδο, το κλειδί κατακερματισμού θα είναι το ίδιο με το κλειδί κατακερματισμού της σωστής εξόδου, και η οποία θα ξεκλειδώσει τη λύση, αλλά είναι αδύνατο να εργαστεί κανείς αντίστροφα και από το κλειδί κατακερματισμού να καταλάβει ποιά θα πρέπει να είναι πραγματικά η έξοδος. Ο κατακερματισμός χρησιμοποιείται για τη δημιουργία ψηφιακών υπογραφών για έγγραφα και απαιτεί λίγο περισσότερη δουλειά και εξήγηση για τη δημιουργία του, αλλά επιτυγχάνει μια καλή ισορροπία μεταξύ της πρόωρης αποκάλυψης των απαντήσεων με την αποκάλυψή τους όταν πραγματική μπορεί να βοηθήσει.
+
+## Συλλογιστική Υψηλού Επιπέδου {#s:exercises-higher}
+
+Υπάρχουν πολλοί άλλοι τύποι ασκήσεων προγραμματισμού ο οποίοι είναι εξίσου δύσκολο να αξιολογηθούν τόσο για τους εκπαιδευτικούς σε μια τάξη μερικών δεκάδων μαθητών και όσο και για τις αυτοματοποιημένες πλατφόρμες βαθμολόγησης. Μεγαλύτερα εργασίες προγραμματισμού, ή προγράμματα στα οποία οι μαθητές θέτουν τους δικούς τους στόχους, (ελπίζουμε ότι) είναι η γενικότερη κατεύθυνση στην οποίο πηγαίνουν τα μαθήματα προγραμματισμού. Οι συζητήσεις ελεύθερης μορφής ή η αιφνίδια συγγραφή κώδικα (twitch-coding) ([s:performance-live](#SECTION)) είναι επίσης πολύτιμες μορφές ασκήσεων, αλλά δεν μπορούν να γίνουν σε μεγάλη κλίμακα.
```

Comment:
τόσο για--> τόσο από
και όσο και για--> όσο και από
Μεγαλύτερα--> Μεγαλύτερες
στην οποίο-->στην οποία

Reply:
Τhanks @pdelias ! Fixed in a new commit.